anvil/sample.py

```diff
@@ -34,64 +148,72 @@ def sample(loaded_model, action, target_length: int, n_large=20000) -> torch.Ten
         acceptance condition
     target_length: int
         the desired number of states to generate from the loaded_model
-    n_large: int (default 20000)
-        the number of total states to generate from the loaded_model per batch, set
-        to a reasonable default value. setting this number excessively high can
-        cause slow down due to running out of memory
+    tskip_guess: int
+        an initial guess for the subsampling interval required to produce
+        decorrelated configurations. By default, this is computed automatically
+        using the first batch of configurations.
 
     Returns
     -------
     sample: torch.Tensor
         a sample of states from loaded_model, size = (target_length, loaded_model.size_in)
 
     """
-    accepted = 0
-    rejected = 0
-    batches = torch.empty((target_length, loaded_model.size_in), dtype=torch.float32)
-
-    for batch in range(ceil(target_length/n_large)):
-        batch_length = min(n_large, target_length-n_large*batch)
-
-        with torch.no_grad():  # don't track gradients
-            z = torch.randn((batch_length, loaded_model.size_in))  # random z configurations
-            phi = loaded_model.inverse_map(z)  # map using trained loaded_model to phi
-            log_ptilde = loaded_model(phi)
-        accept_reject = torch.zeros(batch_length, dtype=bool)
-        chain_indices = torch.zeros(batch_length, dtype=torch.long)
-
-        log_ratio = log_ptilde + action(phi)
-        if not isfinite(exp(float(min(log_ratio) - max(log_ratio)))):
-            raise ValueError("could run into nans")
-
-        if batch == 0:  # first batch
-            log_ratio_i = log_ratio[0]
-            state_i = phi[0]
-        else:
-            log_ratio[0] = log_ratio_i # else set first to be last accepted
-            phi[0] = state_i
-        i = 0
-        for j in range(1, batch_length):
-            condition = min(1, exp(float(log_ratio_i - log_ratio[j])))
-            if np.random.uniform() <= condition:
-                accept_reject[j] = True
-                chain_indices[j] = j
-                log_ratio_i = log_ratio[j]
-                state_i = phi[j]
-                accepted += 1
-                i = j
-            else:
-                accept_reject[j] = False
-                chain_indices[j] = i
-                rejected += 1
-        start = n_large*batch
-        end = start + batch_length
-        batches[start:end, :] = phi[chain_indices, :]
+    batch_length = min(target_length, 10000)  # hard coded for now
```

Comment:
fair enough, although I think we could have just kept `n_large=10000` in the function arguments, just incase somebody does have memory issues (might be an issue as we scale `lattice_length`)

Reply:
btw it can have a different name, maybe `batch_size` in general would work well - and make runcard configuration cleaner